enumerable_methods.rb.rb

```diff
@@ -0,0 +1,100 @@
+module Enumerable
+  def my_each
+    array = self
+    array.size.times { |i| yield array[i] }
+    array
+  end
+
+  def my_each_with_index
+    array = self
+    array.length.times { |i| yield array[i], i }
+    array
+  end
+
+  def my_select
+    array = self
+    my_select_array = []
+    array.my_each { |item| my_select_array << item if yield(item) }
+    my_select_array
+  end
+
+  def my_all?
+    array = self
+    i = 0
+    all = true
+    while i < array.length
+      unless yield(array[i])
+        all = false
+        break
+      end
+      i += 1
+    end
+    all
+  end
+
+  def my_any?
+    array = self
+    i = 0
+    any_item = false
+    while i < array.length
+      if yield(array[i])
+        any_item = true
+        break
+      end
+      i += 1
+    end
+    any_item
+  end
+
+  def my_none?
+    array = self
+    i = 0
+    no_item = true
+    while i < array.length
+      if yield (array[i])
+        no_item = false
+        break
+      end
+      i += 1
+    end
+    no_item
+  end
+
+  def my_count
+    array = self
+    i = 0
+    array.my_each { |item| i += 1 if yield(item) }
+    i
+  end
+
+  def my_inject
+    array = self
+    injected = 0
+    (array.size - 1).times do |i|
+      self[i + 1] = yield(self[i], self[i.next])
+      injected = self[i + 1]
+    end
+    injected
+  end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- When you pass a block to perform an operation, your method is not given the right result. 

- Your method should work with symbols as arguments (:+) and perform the right operation.